lib/http_server.ex

```diff
@@ -1,18 +1,33 @@
 defmodule HTTPServer do
-  @moduledoc """
-  Documentation for `HTTPServer`.
-  """
+  require Logger
 
-  @doc """
-  Hello world.
+  def accept(port) do
+    tcp_options = [:binary, {:packet, 0}, {:active, false}, reuseaddr: true]
+    {:ok, socket} = :gen_tcp.listen(port, tcp_options)
+    Logger.info("Accepting connections on port #{port}")
+    listen(socket)
+  end
+
+  defp listen(socket) do
+    {:ok, client} = :gen_tcp.accept(socket)
+    spawn(fn -> serve(client) end)
+    listen(socket)
+  end
 
-  ## Examples
+  defp serve(socket) do
+    socket
+    |> read_line()
+    |> write_line(socket)
 
-      iex> HTTPServer.hello()
-      :world
+    serve(socket)
+  end
+
+  defp read_line(socket) do
+    {:ok, data} = :gen_tcp.recv(socket, 0)
+    data
+  end
 
-  """
-  def hello do
-    :world
+  defp write_line(data, socket) do
+    :gen_tcp.send(socket, data)
   end
```

Comment:
```suggestion
  defp write_line(socket, data) do
    :gen_tcp.send(socket, data)
  end
```

Reply:
I was thinking about this, and it introduces a bit of a puzzle as to how to capture the data.  I think this will be solved once you figure out encodings, as your patterns will have to change.  Up to you if you want to work on this now or later.